### DISCUSSION TITLE
exploring-dotnet-6-part-10-new-dependency-injection-features-in-dotnet-6/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#98] [Announcements]
# exploring-dotnet-6-part-10-new-dependency-injection-features-in-dotnet-6/

In this post I talk about some of the new features added to the DI libraries in .NET 6, as well a performance feature that didn't make it in.

https://andrewlock.net/exploring-dotnet-6-part-10-new-dependency-injection-features-in-dotnet-6/

**Comment by dbeylkhanov** [comment]
IServiceProviderIsService looks like implementaton of Service Locator anti-pattern

**Reply by meJevin** [reply]
Почему? Где ты это увидел?

**Reply by meJevin** [reply]
Сервис локатор это про динамическое создание зависимостей класс внутри самого класса с помощью глобальных классов/переменных. 
Тут речь про то, что под капотом у DI контейнера дотнетовского. Конечно он будет знать много про все типы, с которыми он работает. Иначе, как ему выполнгять роль DI контейнера)
Короче, я вообще связи не вижу между сервис локатором и кишками DI дотнета

[a comment by an automated account is not shown]
